### PR TITLE
[tp] Implement reidentifySource()

### PR DESCRIPTION
**Pull request by zhoonit**
## Commits to be reviewed in this PR

<details><summary>[tp] Implement reidentifySource()</summary><br />

This patch implements reidentifySource(). Test will be followed

**Self evaluation:**
1. Build test: [X]Passed [ ]Failed [ ]Skipped
2. Run test: [X]Passed [ ]Failed [ ]Skipped

Signed-off-by: Jihoon Lee <jhoon.it.lee@samsung.com>

</details>

### CODE REVIEW

File: nntrainer/tensor/tensor_pool.cpp [hunk 1]
```diff
@@ -351,6 +351,48 @@ Tensor *TensorPool::requestOrExtend(const std::string &name,
   }
 }
 
+void TensorPool::reidentifySource(const std::string &dest,
```
Comment (jijoongmoon) [inline nntrainer/tensor/tensor_pool.cpp:354]: We need another func to support other cases we discussed off-line.

**Review by jijoongmoon — APPROVED** [review]
LGTM

**Review by kparichay — APPROVED** [review]
LGTM

**Review by kparichay — COMMENTED** [review]
It would be good to have some unittests for this :)

**Pull request merged**